lib/portlayer/storage/vsphere/volume.go

```diff
@@ -51,11 +51,17 @@ func NewVolumeStore(op trace.Operation, storeName string, s *session.Session, ds
 		return nil, err
 	}
 
-	dm, err := disk.NewDiskManager(op, s, DetachAll)
+	dm, err := disk.NewDiskManager(op, s)
 	if err != nil {
 		return nil, err
 	}
 
+	if DetachAll {
```

Comment:
Agreed but this is so tangled right now so I rather not to touch it but keep the current logic. The bigger issue is the storage rest api is not simply acting like a proxy but doing a lot more than it. They call NewVolumeStore in the configure phase  which is called by portlayer Init during the initialization.